src/platform/datapath_raw_socket.c

```diff
@@ -516,19 +516,24 @@ CxPlatFramingTransportChecksum(
     _In_ uint32_t AddrLength,
     _In_ uint16_t NextHeader,
     _In_reads_(IPPayloadLength) uint8_t* IPPayload,
-    _In_ uint32_t IPPayloadLength
+    _In_ uint32_t IPPayloadLength,
+    _In_ BOOLEAN PseudoHeaderOnly
```

Comment:
What's this for?

Reply:
HW xsum offload requires a pseudoHeader to be passed down to the NIC

Reply:
`PseudoHeaderOnly` computes the framing checksum's IP pseudoheader checksum without computing the final checksum over the transport layer header and data.

Reply:
> HW xsum offload requires a pseudoHeader to be passed down to the NIC

Sorry, I missed this answer when writing mine. BTW, from what I could tell when skimming DPDK, the pseudoheader might or might not need to be filled in, depending on the PMD. We don't have any tests yet, but I think we have a latent bug in our own DPDK code, because we need to conditionally zero/initialize with a pseudoheader checksum for DPDK.